src/common/exif.cc

```diff
@@ -3988,8 +4005,31 @@ int dt_exif_xmp_attach_export(const int imgid, const char *filename, void *metad
   }
   catch(Exiv2::AnyError &e)
   {
-    std::cerr << "[dt_exif_xmp_attach_export] " << filename << ": caught exiv2 exception '" << e << "'\n";
-    return -1;
+    std::string s(e.what());
+    if(s == "Size of XMP JPEG segment is larger than 65535 bytes")
```

Comment:
That's not how you detect type of exception, especially since the message might be translated. Instead you should check e.code() == Exiv2::kerTooLargeJpegSegment.

Reply:
Excellent. I'll do.